GeneratorInterface/GenFilters/plugins/PhotonGenFilter.cc

```diff
@@ -0,0 +1,78 @@
+#include "GeneratorInterface/GenFilters/plugins/PhotonGenFilter.h"
+#include "SimDataFormats/GeneratorProducts/interface/HepMCProduct.h"
+#include "GeneratorInterface/GenFilters/plugins/MCFilterZboostHelper.h"
+#include <iostream>
+
+using namespace edm;
+using namespace std;
+
+PhotonGenFilter::PhotonGenFilter(const edm::ParameterSet &iConfig)
+    : token_(consumes<edm::HepMCProduct>(
+          edm::InputTag(iConfig.getUntrackedParameter("moduleLabel", std::string("generator")), "unsmeared"))) {
+  // Constructor implementation
+  ptMin = iConfig.getUntrackedParameter<double>("MinPt", 20.);
+  etaMin = iConfig.getUntrackedParameter<double>("MinEta", -2.4);
+  etaMax = iConfig.getUntrackedParameter<double>("MaxEta", 2.4);
+  drMin = iConfig.getUntrackedParameter<double>("drMin", 0.1);
+  ptThreshold = iConfig.getUntrackedParameter<double>("ptThreshold", 2.);
+}
+
+PhotonGenFilter::~PhotonGenFilter() {
+  // Destructor implementation
+}
+
+bool PhotonGenFilter::filter(edm::StreamID, edm::Event &iEvent, const edm::EventSetup &iSetup) const {
+  using namespace edm;
+  Handle<HepMCProduct> evt;
+  iEvent.getByToken(token_, evt);
+  const HepMC::GenEvent *myGenEvent = evt->GetEvent();
+
+  for (HepMC::GenEvent::particle_const_iterator p = myGenEvent->particles_begin(); p != myGenEvent->particles_end();
+       ++p) {
+    if ((*p)->pdg_id() == 22) {
+      if ((*p)->momentum().perp() > ptMin && (*p)->status() == 1 && (*p)->momentum().eta() > etaMin &&
+          (*p)->momentum().eta() < etaMax) {
+        bool accepted_photon = true;
+        double phi = (*p)->momentum().phi();
+        double eta = (*p)->momentum().eta();
+        for (HepMC::GenEvent::particle_const_iterator q = myGenEvent->particles_begin();
+             q != myGenEvent->particles_end();
+             ++q) {
+          if (&p != &q) {
+            if ((*q)->momentum().perp() > ptThreshold && (*q)->pdg_id() != 22 &&
```

Comment:
a general concern on the logic is that: will this filter select photon from pi0 decay and it will pass the filter since you only require object_pdgid!=22, I would suggest you do a further study to check whether a pi0 exclusion will promote your desired event production or not